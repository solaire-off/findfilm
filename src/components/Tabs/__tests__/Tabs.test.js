import React from "react";
import { screen, render } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { Tabs } from "..";

const selectedTab = "All";
const tabsList = [
  {
    name: selectedTab,
  },
  {
    name: "Another tab",
  },
];

it("Current render", () => {
  const { asFragment } = render(
    <Tabs list={tabsList} selectedTab={selectedTab} callback={() => {}} />
  );

  expect(asFragment()).toMatchSnapshot();
});

it("Callback onClick work", () => {
  const mockCallback = jest.fn();
  render(
    <Tabs list={tabsList} selectedTab={selectedTab} callback={mockCallback} />
  );

  userEvent.click(screen.getByText("Another tab"));
  expect(mockCallback).toBeCalledTimes(1);
});
