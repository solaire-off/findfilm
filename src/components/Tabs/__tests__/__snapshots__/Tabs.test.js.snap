// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Current render 1`] = `
<DocumentFragment>
  <div
    class="tabs-list"
  >
    <button
      class="tabs-list__item tabs-list__item--active"
      type="button"
    >
      All
    </button>
    <button
      class="tabs-list__item "
      type="button"
    >
      Another tab
    </button>
  </div>
</DocumentFragment>
`;
